nautobot/core/testing/__init__.py

```diff
@@ -69,6 +75,38 @@ def run_job_for_testing(job, username="test-user", **kwargs):
     return job_result
 
 
+def create_job_result_and_run_job(module, name, source="local", *args, **kwargs):
```

Comment:
I know `run_job_for_testing` is already defined directly in `__init__.py`, but should we consider putting these functions in a new submodule like `nautobot/core/testing/jobs.py` and just importing them here if desired?

Reply:
I still want to do this. Was focused on the underlying issues with triggering sync/dry-run first.